src/vsr.zig

```diff
@@ -1097,6 +1097,47 @@ test "quorums" {
     }
 }
 
+/// Deterministically assigns replica_ids for the initial configuration.
+///
+/// Eventualy, we want to identify replicas using random u128 ids to prevent operator errors.
+/// However, that requires unergonomic two-step process for spinning a new cluster up.  To avoid
+/// needlessly compromizing the experience until reconfiguration is fully implemented, derive
+/// replica ids for the initial cluster deterministically.
+pub fn initial_configuration(cluster: u32) [constants.nodes_max]u128 {
+    const IdSeed = packed struct {
+        cluster: u32,
+        replica: u8,
+    };
+
+    var result = [_]u128{0} ** constants.nodes_max;
+    var replica: u8 = 0;
+    while (replica < constants.nodes_max) : (replica += 1) {
+        result[replica] = checksum(std.mem.asBytes(&IdSeed{ .cluster = cluster, .replica = replica }));
+    }
+
+    assert_valid_configuration(&result);
+    return result;
+}
+
+/// Check that:
+///  - all non-zero elements are different
+///  - all zero elements are trailing
+pub fn assert_valid_configuration(configuration: *const [constants.nodes_max]u128) void {
+    for (configuration) |replica_i, i| {
+        for (configuration[0..i]) |replica_j| {
+            if (replica_j == 0) assert(replica_i == 0);
```

Comment:
I do plan to encode length into the configuration by just zeroing out suffix, but in this implementaiton configurations _always_ contain nodes_max entries. This is because we don't want to pass `--standby-count` to `format`, so we just "pre-allocate" all standbys ids. 

Even if we keep "initial cluster memebres get deterministic ids" trick, long term standbys would definitely use random ids, and we would just forbid standbys in the initial config. But, as we can't add sbys dynamically yet, we preallocate instaed.

Reply:
> So to see if I understand: Right now initial_configuration creates ids for every possible standby, even though the initial configuration is not allowed any standbys?

Not exactly, it is quite a bit less elegant. 

In this pr, the root configuration looks as if we have a maximum number of standbys configured. However, we ignore the number of standbys implicitly encoded in the configuration, and instead derive that from `--addresses` passed to `.start`.

This is hacky, but avoids the need to encode the number of standbys during format. So, you can format the cluster, start it with 3 standbys, then restart with zero standbys, and that'll work, as the standby count right now is purely in-memory thing. 

But the bigger reason is that before this PR we don't pass `--standby-count` to format, and I want to keep it that way, just to avoid having to adjust user-visible interface while it is still in flux. 

---

Long term, I do want to rely on `superblock.vsr_state.configuration` to tell me the number of standbys, but, to make that useful, we need to have an API to actually change standys dynamically.